ecosystem/sep-0008.md

```diff
@@ -225,17 +225,44 @@ Name | Type | Description
 `status` | string | `"action_required"`
 `message` | string | A human readable string containing information regarding the action required.
 `action_url` | string | A URL that allows the user to complete the actions required to have the transaction approved.
+`action_method` | string | (Optional) `GET` or `POST`, indicating the type of request that should be made to the `action_url`. If not provided, `GET` is assumed.
+`action_fields` | string[] | (Optional) An array of additional fields defined by [SEP-9](sep-0009.md) that the client may optionally provide to the approval service when sending the request to the `action_url`.
 
 ###### Example
 
 ```json
 {
   "status": "action_required",
   "message": "Please sign the terms of service via the provided URL.",
-  "action_url": "https://..."
+  "action_url": "https://...",
+  "action_method": "POST",
+  "action_fields": ["email_address", "mobile_number"]
 }
 ```
 
+###### Action URL
+If the `action_method` is provided the client uses the type of request indicated when sending a request to the `action_url`. If `action_fields` are to be provided and the fields contain sensitive or personal information the `POST` case is preferred to ensure sensitive information is not transmitted in the URL.
+
+- In the case that `action_method` is `GET` any fields requested may be passed as query parameters in the `action_url`.
+
+- In the case that `action_method` is `POST` any fields requested may be passed as JSON in the request body, using content type `application/json`. The server should respond to the `POST` with a HTTP response with status code `200`, content-type `application/json` and a body containing the next URL that the user should be taken to.
+
+   The next URL in the POST case is necessary to support products that are websites and products that are mobile applications. Products that run entirely in the browser can redirect the user to the next URL and the redirect prevents odd user experiences where a user refreshes a page and is prompted to re-post the data. Products that are mobile applications cannot open a system browser with a POST request. They can make the POST request internally, extract the redirect in the response, and open a system browser with the next URL.
+
+   Parameters:
+
+   Name | Type | Description
+   -----|------|------------
+   `next_url` | string | A URL where the user can complete the required actions with all the parameters included in the original POST pre-filled or already accepted.
+
+   Example:
+
+   ```json
+   {
+     "next_url": "https://..."
+   }
+   ```
```

Comment:
This section provides clear instruction for what the action_url's response message should contain, but is not clear about how the request should be formatted or constructed. There are a couple different approaches that an implementer could take, either using SEP-12 for inspiration, or using each SEP-9 field as a key in a flat object, or using each SEP-9 field as a hierarchy.

@shredding @dmgmyza Could you share how the request is formatted in your implementation?

I'll update this PR with the additional details once I hear from you.

Reply:
I've opened PR https://github.com/DSTOQ/stellar-protocol/pull/3 that contains additional details.